src/main/java/creativei/enums/AreaOfInterest.java

```diff
@@ -14,7 +15,7 @@ private AreaOfInterest(String value) {
         this.value =value;
     }
 
-    private static final Map<String, AreaOfInterest> lookup = new HashMap<String, AreaOfInterest>();
+    private static final Map<String, AreaOfInterest> lookup = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
No need to use TreeMap. HashMap is good enough

Reply:
tree map provides directly string comparison with ignoring case .In hash map i have to convert both string either in lower case or upper case to compare . Tree Map complexity in worst case is O(logn) whereas hashmap has O(n)